Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -686,10 +695,11 @@ private extension Manifest {
     /// Returns the names of all the referenced modules in the manifest.
     func allReferencedModules() -> Set<String> {
         let names = package.targets.flatMap { target in
-            [target.name] + target.dependencies.map {
+            [target.name] + target.dependencies.flatMap {
                 switch $0 {
                 case .Target(let name):
                     return name
+                default: return nil
```

Comment:
I think it might be better to have specific cases for `.ByName` and `.Product` here to return `nil`, so that the compiler will remind us to handle any possible fourth type of reference in the future.